File: java/src/test/java/org/example/algorithms/two_pointer/PalindromeTest.java
package org.example.algorithms.two_pointer;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PalindromeTest {
    @Test
    public void testIsPalindrome() {
        Palindrome palindrome = new Palindrome();

        assertTrue(palindrome.isPalindrome("A man, a plan, a canal: Panama"));
        assertFalse(palindrome.isPalindrome("race a car"));
        assertTrue(palindrome.isPalindrome(" "));
        assertTrue(palindrome.isPalindrome("a."));
        assertFalse(palindrome.isPalindrome("0P"));
    }
}

File: go/algorithms/two_pointer/palindrome_test.go
package twopointer

import "testing"

func TestIsPalindrome(t *testing.T) {
	tests := []struct {
		input    string
		expected bool
	}{
		{"A man, a plan, a canal: Panama", true},
		{"race a car", false},
		{"hello", false},
		{"", true},
		{"a", true},
		{"ab", false},
		{"madam", true},
		{"step on no pets", true},
	}

	for _, test := range tests {
		result := IsPalindrome(test.input)
		if result != test.expected {
			t.Errorf("IsPalindrome(%q) = %v; want %v", test.input, result, test.expected)
		}
	}
}
